### PR TITLE
drivers: wifi: FMAC API documentation update

### PR DESCRIPTION
**Pull request by VivekUppunda**
This change is to update the doxygen documentation for FMAC API.

### CODE REVIEW

**Review by krish2718 — APPROVED** [review]

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h [hunk 1]
```diff
@@ -35,15 +35,19 @@
 /**
  * @brief Initializes the UMAC IF layer.
  *
- * @param data_config Pointer to configuration of data queues.
+ * @param data_config  Pointer to configuration of data queues.
```
[the remark this answers is not in the document]
Comment (VivekUppunda) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h:38]: completer

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h [hunk 2]
```diff
@@ -35,15 +35,19 @@
...
  * @param rx_buf_pools Pointer to configuration of Rx queue buffers.
  *		       See rx_buf_pool_params
- * @param callbk_fns Pointer to callback functions.
+ * @param callbk_fns   Pointer to callback functions for addressing events
```
[the remark this answers is not in the document]
Comment (VivekUppunda) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h:41]: completed

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h [hunk 3]
```diff
@@ -35,15 +35,19 @@
...
+ *                     from the UMAC layer. e.g. callback function to process
+ *                     packet received from lower layer, scan result etc
```
[the remark this answers is not in the document]
Comment (VivekUppunda) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h:43]: completed

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h [hunk 4]
```diff
@@ -35,15 +35,19 @@
...
  *
  * This function initializes the UMAC IF layer.It does the following:
  *	    - Creates and initializes the context for the UMAC IF layer.
  *	    - Initializes the HAL layer.
- *	    - Registers the driver to the underlying OS.
+ *	    - Initializes the OS abstraction Layer
+ *	    - Initializes TX queue token sizes
+ *	    - Initializes the RX Buffer Pool
```
[the remark this answers is not in the document]
Comment (VivekUppunda) [reply drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h:50]: completed

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h [hunk 5]
```diff
@@ -653,10 +705,11 @@ enum wifi_nrf_status wifi_nrf_fmac_get_station(void *fmac_dev_ctx,
  * @param dev_ctx Pointer to the UMAC IF context for a RPU WLAN device.
  * @param if_idx VIF index.
  *
- * This function is used to send a command
- *	    to get interface statistics using interface index.
+ * This function is used to send a command to:
+ *	    - Get interface statistics using interface index.
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h:709]:
```suggestion
 *	    - Get interface statistics using interface index \p if_idx.
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h [hunk 6]
```diff
@@ -668,10 +721,11 @@ enum wifi_nrf_status wifi_nrf_fmac_get_interface(void *dev_ctx,
  * @param if_idx Index of the interface on which power management is to be set.
  * @param state Enable/Disable of WLAN power management.
  *
- * This function is used to send a command
- *	    to RPU to Enable/Disable WLAN Power management.
+ * This function is used to send a command to:
+ *	     - the RPU firmware to Enable/Disable WLAN Power management.
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h:725]:
```suggestion
 *	     - The RPU firmware to enable/disable WLAN power management.
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h [hunk 7]
```diff
@@ -741,9 +799,10 @@ enum wifi_nrf_status wifi_nrf_fmac_set_wowlan(void *fmac_dev_ctx,
  * @param fmac_dev_ctx Pointer to the UMAC IF context for a RPU WLAN device.
  * @param if_idx Index of the interface on which the CMD needs to be sent.
  *
- * This function is used to get PHY configuration from RPU.
+ * This function is used to get PHY configuration from RPU firmware.
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h:802]:
```suggestion
 * This function is used to get PHY configuration from the RPU firmware.
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h [hunk 8]
```diff
@@ -766,10 +826,11 @@ enum wifi_nrf_status wifi_nrf_fmac_register_frame(void *fmac_dev_ctx, unsigned c
  * @param if_idx Index of the interface on which the CMD needs to be sent.
  * @param wiphy_info wiphy parameters
  *
- * This function is used to send a command
- *	    to RPU to configure parameters related to interface.
+ * This function is used to send a command to the RPU firmware to:
+ *	    - To configure parameters related to interface.
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h:830]:
```suggestion
 *	    - To configure interface specific parameters on an interface identified by \p if_idx..
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h [hunk 9]
```diff
@@ -782,10 +843,11 @@ enum wifi_nrf_status wifi_nrf_fmac_set_wiphy_params(void *fmac_dev_ctx,
  * @param if_idx Index of the interface on which the TWT parameters be set.
  * @param twt_info TWT parameters
  *
- * This function is used to send a command
- *	    to RPU to configure parameters related to TWT setup.
+ * This function is used to send a command to the RPU firmware to:
+ *	    - Configure parameters related to TWT setup.
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h:847]:
```suggestion
 *	    - Configure TWT setup specific parameters.
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h [hunk 10]
```diff
@@ -797,24 +859,26 @@ enum wifi_nrf_status wifi_nrf_fmac_twt_setup(void *fmac_dev_ctx,
  * @param if_idx Index of the interface on which the TWT parameters be set.
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h:859]:
```suggestion
 * @param if_idx Index of the interface on which the TWT parameters are to be set.
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h [hunk 11]
```diff
@@ -853,9 +917,13 @@ void wifi_nrf_fmac_dev_rem(struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx);
  * @param op_band Operating band of the RPU.
  * @param tx_pwr_ctrl_params TX power control parameters to be passed to the RPU.
  *
- * This function initializes the firmware of an RPU instance.
+ * This function initializes the firmware of an RPU instance. The following is addressed
+ *              - BAL layer device initialization
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h:921]:
```suggestion
 *              - HAL layer device initialization
 *              - BAL layer device initialization
```

File: drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h [hunk 12]
```diff
@@ -853,9 +917,13 @@ void wifi_nrf_fmac_dev_rem(struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx);
...
+ *              - FW initialization and phy calibration data is sent to PHY
```
Comment (sachinthegreen) [inline drivers/wifi/nrf700x/osal/fw_if/umac_if/inc/default/fmac_api.h:922]:
```suggestion
 *              - FW initialization and PHY calibration data is sent to PHY
```

**Review by sachinthegreen — APPROVED** [review]

**Comment by VivekUppunda** [comment]
the CI run is failing for the check for test-sdk-wifi-EK-shell. CI failure reason is attached. As per discussion in NCS-CI chat - this is to be expected. 

[Yesterday 2:42 PM] Gadde, KranthiWi-Fi test-sdk-wifi-EK-shell/master CI Issue
Hi NCS CI .. If you find any Wi-Fi CI failing only because of test-sdk-wifi-EK-shell, then please ignore this . Recently we have merged this under a single Jenkinsplan now and is named as test-sdk-wifi,.. and we have removed this job totally .Tests will trigger on DK and EK using a single Jenkins plan ... If any of your PRs are failing only with this reason, request to please post in PR merge group mentioning this reason as status is correlated with given SHA.

please merge this PR .

Thanks and regards
Vivek
PR-CI-Run-failure

**Pull request merged**